source/connection-monitoring-and-pooling/tests/pool-checkout-returned-connection-maxConnecting.yml

```diff
@@ -0,0 +1,106 @@
+version: 1
+style: integration
+description: threads blocked by maxConnecting check out returned connections
+runOn:
+  -
+    # required for blockConnection in fail point
+    minServerVersion: "4.4.0"
+failPoint:
+  configureFailPoint: failCommand
+  mode: { times: 3 }
+  data:
+    failCommands: ["isMaster"]
+    closeConnection: false
+    blockConnection: true
+    blockTimeMS: 750
+poolOptions:
+  maxPoolSize: 10
+  waitQueueTimeoutMS: 5000
+operations:
+  # start a thread, check out a connection, and
+  # hold it until the other threads are blocked
+  # on maxConnecting.
+  - name: start
+    target: thread1
+  - name: checkOut
+    thread: thread1
+    label: conn0
+  # Start 2 new threads that wait for thread1 to check
+  # out its connection before attempting to check out their own.
+  # This fills maxConnecting for the next 750ms.
```

Comment:
Why do these threads need to call waitForEvent? Can we instead wait for ConnectionCheckedOut on the main thread before starting these threads?
```
  - name: waitForEvent
    event: ConnectionCheckedOut
    count: 1
# Start 2 new threads to fill maxConnecting for the next 750ms.
...
```

Reply:
Done